src/helpers/format.ts

```diff
@@ -306,8 +312,24 @@ function formatJSTime(jsDate: Date, format: Format): FormattedValue {
 // CREATE / MODIFY FORMAT
 // -----------------------------------------------------------------------------
 
+/**
+ * Create a default format for a number.
+ *
+ * If possible this will try round the number to have less than DEFAULT_FORMAT_NUMBER_OF_DIGITS characters
+ * in the number. This is obviously only possible for number with a big decimal part. For number with a lot
+ * of digits in the integer part, keep the number as it is.
+ */
 export function createDefaultFormat(value: number): Format {
-  let { decimalDigits } = splitNumber(value, 10);
+  let { integerDigits, decimalDigits } = splitNumber(value);
+  if (!decimalDigits) return "0";
+
+  // If there's no space for at least the decimal separator + a decimal digit, don't clip the number
+  if (integerDigits.length + 2 > DEFAULT_FORMAT_NUMBER_OF_DIGITS) {
+    return "0." + "0".repeat(decimalDigits.length);
```

Comment:
functional question: why not return "0" ?

Reply:
I think that it's worth for the user to have the information that it's a decimal number, and not just an integer. 

But maybe not, I though about having "0" too, I wasn't too sure which one was better

Reply:
I understand,

In this particular case, there will be a lot of digits on the integer part. 
The decimal part will therefore be significantly not important.

Maybe keep the simple rule of a maximum number of digits by default, in all cases :)

Reply:
Changed to not display decimals if we have no space for the decimal separator + a decimal digit.